components/add_user_to_channel_modal/add_user_to_channel_modal.jsx

```diff
@@ -0,0 +1,316 @@
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import {getFullName} from 'mattermost-redux/utils/user_utils';
+
+import SearchChannelWithPermissionsProvider from 'components/suggestion/search_channel_with_permissions_provider.jsx';
+import SuggestionBox from 'components/suggestion/suggestion_box.jsx';
+import SuggestionList from 'components/suggestion/suggestion_list.jsx';
+
+import {placeCaretAtEnd} from 'utils/utils.jsx';
+
+export default class AddUserToChannelModal extends React.Component {
+    static propTypes = {
+
+        /**
+         * Function that's called when modal is closed
+         */
+        onHide: PropTypes.func.isRequired,
+
+        /**
+         * The user that is being added to a channel
+         */
+        user: PropTypes.object.isRequired,
+
+        /**
+         * Object used to determine if the user
+         * is a member of a given channel
+         */
+        channelMembers: PropTypes.object.isRequired,
+
+        actions: PropTypes.shape({
+
+            /**
+             * Function to add the user to a channel
+             */
+            addChannelMember: PropTypes.func.isRequired,
+
+            /**
+             * Function to fetch the user's channel membership
+             */
+            getChannelMember: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+
+            /**
+             * Whether or not the modal is visible
+             */
+            show: true,
+
+            /**
+             * Whether or not a request to add the user is in progress
+             */
+            saving: false,
+
+            /**
+             * Whether or not a request to check for the user's channel membership
+             * is in progress
+             */
+            checkingForMembership: false,
+
+            /**
+             * The user input in the channel search box
+             */
+            text: '',
+
+            /**
+             * The id for the channel that is selected
+             */
+            selectedChannelId: null,
+
+            /**
+             * An error to display when the add request fails
+             */
+            inviteError: '',
+        };
+        this.suggestionProviders = [new SearchChannelWithPermissionsProvider()];
+        this.enableChannelProvider();
+    }
+
+    enableChannelProvider = () => {
+        this.suggestionProviders[0].disableDispatches = false;
+    }
+
+    focusTextbox = () => {
+        if (this.channelSearchBox == null) {
+            return;
+        }
+
+        const textbox = this.channelSearchBox.getTextbox();
+        if (document.activeElement !== textbox) {
+            textbox.focus();
+            placeCaretAtEnd(textbox);
+        }
+    }
+
+    onChange = (e) => {
```

Comment:
I feel like this method could have a more descriptive name. What is changing to trigger this event?

Reply:
I agree. It is called when the user types a character into the input textbox, so I'm changing the name to `onInputChange()`.